src/service_objects/game_piece_action_resolvers/attack_resolver.ts

```diff
@@ -0,0 +1,53 @@
+import * as Types from '../../graphql/__generated__/resolvers-types';
+import * as Models from '../../models/index.js';
+import { diceRoll } from '../../graphql/helpers';
+import { EncodedDiceRolls, ResolvedAttack } from '../../models/game_piece_action';
+
+export default function resolveAttacks(
+  attackingUnit: Models.Unit,
+  targetUnit: Models.Unit,
+  encodedDiceRolls: EncodedDiceRolls
+): ResolvedAttack[] {
+  // TODO: Decode dice roll results using private key
+  //  For now just simulate rolls here
+  const decodedDiceRolls = simulateDiceRolls(attackingUnit.rangedNumAttacks);
+
+  // Gather details of each attack and determine damage
+  let attacks = [];
+  let totalDamage = 0;
+  for (var i = 0; i < attackingUnit.rangedNumAttacks; i++) {
+    const rollsOffset = i * 3;
+    const hitRoll = decodedDiceRolls[rollsOffset];
+    const woundRoll = decodedDiceRolls[rollsOffset + 1];
+    const saveRoll = decodedDiceRolls[rollsOffset + 2];
+
+    const hitRollSuccess = hitRoll >= attackingUnit.rangedHitRoll;
+    const woundRollSuccess = woundRoll >= attackingUnit.rangedWoundRoll;
+
+    const armorPiercing = attackingUnit.rangedArmorPiercing || 0;
+    const modifiedSave = targetUnit.armorSaveRoll + armorPiercing;
+    const saveRollSuccess = saveRoll >= modifiedSave;
+
+    const damageDealt = hitRollSuccess && woundRollSuccess && !saveRollSuccess ? attackingUnit.rangedDamage : 0;
+
+    attacks.push({
```

Comment:
These would be nice to push to something like New Relic if we get the chance.  Mina folks would love to hear if we process like a thousand turns, and we can show a little chart of how lucky people were and stuff in our final pitch.

Reply:
Agreed!